src/redis/hyperloglog.c

```diff
@@ -1301,9 +1301,10 @@ int convertSparseToDenseHll(struct HllBufferPtr in_hll, struct HllBufferPtr out_
   return C_OK;
 }
 
-int pfadd_sparse(sds* hll_ptr, unsigned char* value, size_t size, int* promoted) {
+int pfadd_sparse(sds* hll_ptr, const unsigned char* value,
```

Comment:
changed the interface to accept const unsigned char. unfortunately the internal utilities still require mutable buffer, so I remove const below.